### PR TITLE
StructType field  `optional` by default

### PR DESCRIPTION
**Pull request by MehulBatra**
* Type change default = false
* Updated test cases to align with recent changes in the codebase (Unit + Integration)
* mkdoc changes to update required field default value

### CODE REVIEW

File: pyiceberg/types.py [hunk 1]
```diff
@@ -24,10 +24,10 @@
     ...     NestedField(1, "required_field", StringType(), True),
     ...     NestedField(2, "optional_field", IntegerType())
     ... ))
-    'struct<1: required_field: optional string, 2: optional_field: optional int>'
+    'struct<1: required_field: required string, 2: optional_field: optional int>'
```
Comment (Fokko) [inline pyiceberg/types.py:27]: Nice one :)

File: tests/test_types.py [hunk 2]
```diff
@@ -560,13 +560,13 @@ def test_deserialization_struct() -> None:
 
 
 def test_str_struct(simple_struct: StructType) -> None:
-    assert str(simple_struct) == "struct<1: required_field: required string (this is a doc), 2: optional_field: required int>"
+    assert str(simple_struct) == "struct<1: required_field: required string (this is a doc), 2: optional_field: optional int>"
```
Comment (Fokko) [inline tests/test_types.py:563]: This seems a bit confusing, the name of the field is `required_field`, but it is actually optional.

**Review by Fokko — COMMENTED** [review]
Looks good @MehulBatra Left one small suggestion.

Could you also remove `required=False` from `mkdocs/docs/api.md`. Since this is now default, I think we should remove it to make the docs less verbose.

File: tests/test_types.py (hunk 2 above)
Comment (Fokko) [reply tests/test_types.py:563]: Misread this, sorry about that!

Comment (MehulBatra) [reply tests/test_types.py:563]: acknowledged

**Comment by MehulBatra** [comment]
> Looks good @MehulBatra Left one small suggestion.
> 
> Could you also remove `required=False` from `mkdocs/docs/api.md`. Since this is now default, I think we should remove it to make the docs less verbose.

On it, pushing changes for the same.

**Review by Fokko — APPROVED** [review]
LGTM, thanks for working on this @MehulBatra

File: tests/expressions/test_literals.py [hunk 3]
```diff
@@ -277,8 +277,7 @@ def test_decimal_to_decimal_conversion() -> None:
 def test_timestamp_to_date() -> None:
     epoch_lit = TimestampLiteral(int(datetime.datetime.fromisoformat("1970-01-01T01:23:45.678").timestamp() * 1_000_000))
     date_lit = epoch_lit.to(DateType())
-
-    assert date_lit.value == 0
+    assert abs(date_lit.value) == 0
```
Comment (Fokko) [inline tests/expressions/test_literals.py:280]:
```suggestion

    assert abs(date_lit.value) == 0
```

Comment (Fokko) [reply tests/expressions/test_literals.py:281]:
```suggestion
    assert date_lit.value == 0
```

**Comment by Fokko** [comment]
@MehulBatra I've removed an unrelated change, I think https://github.com/apache/iceberg-python/pull/612/ addresses the issue that you ran into

**Comment by Fokko** [comment]
Thanks for fixing this @MehulBatra Much appreciated!

**Pull request merged**